### ISSUE TITLE
test.js to use own location as base of schema directory.

### ISSUE THREAD

**Issue by RussellMcOrmond** [crkn-rcdr/Access-Platform#21]
Schemas that have includes currently require that the tool be run from the directory of the tool.

`tdr@25309fd0732e:~/Databases$ node ~/Databases/test.js -s schemas/collection.json -f ~/a.json
`

It would be simpler to not have to change directories first.

```
tdr@25309fd0732e:~$ node ~/Databases/test.js -s ~/Databases/schemas/collection.json -f ~/a.json
{ stack:
   'Error: Error opening file "/home/tdr/schemas/_util.json" \nENOENT: no such file or directory, open \'/home/tdr/schemas/_util.json\'\n    at ReadFileContext.fs.readFile [as callback] (/home/tdr/Databases/node_modules/@apidevtools/json-schema-ref-parser/lib/resolvers/file.js:51:20)\n    at FSReqWrap.readFileAfterOpen [as oncomplete] (fs.js:238:13)\n\nError: ENOENT: no such file or directory, open \'/home/tdr/schemas/_util.json\'',
  message:
   'Error opening file "/home/tdr/schemas/_util.json" \nENOENT: no such file or directory, open \'/home/tdr/schemas/_util.json\'',
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/home/tdr/schemas/_util.json',
  toJSON: [Function: toJSON],
  name: 'Error',
  toString: [Function: toString] }
tdr@25309fd0732e:~$ 

```

**Comment by RussellMcOrmond** [comment]
I'm also wondering if the JSON to test could be accepted on STDIN by default.  That way I wouldn't need to create temporary files for testing.

JSON IN, JSON OUT   --- except when there is an error in running tool.

**Comment by SaschaAdler** [comment]
I agree that this is cumbersome, but I'd prefer to have schema testing end up on the API server that will actually be depositing documents into Couch. The way the directory structure needs to be just right in the schema at present is very awkward, and I need to find a solution for it.

Conceivably the API server can even have a series of schema-testing routes, especially when the schemas eventually stabilize.

**Comment by RussellMcOrmond** [comment]
I'm not sure I understand why it would be one or the other (API server or command-line utility).  Seems in development that having both would be better.

As well as not needing to be on a specific path to run (a fairly easy fix), having the ability to point at a Couch URL would be helpful as well.  As it is I'm using curl to download a JSON document, then passing it through the validator.  While there will eventually be an API server, there will be times when we won't want to wait for API server updates (and more complex testing tools) to test new schemas against older documents.

**Comment by SaschaAdler** [comment]
Effectively closed by #44.
